src/client/datascience/jupyter/jupyterDebugger.ts

```diff
@@ -275,12 +277,14 @@ export class JupyterDebugger implements IJupyterDebugger, ICellHashListener {
 
     private buildSourceMap(fileHash: IFileHashes): ISourceMapRequest {
         const sourceMapRequest: ISourceMapRequest = { source: { path: fileHash.file }, pydevdSourceMaps: [] };
-
+        const isIPyKernel6 = this.ipykernelVersion?.toLowerCase().trim().startsWith('6') === true;
```

Comment:
We'd have to remember to update this when ipykernel 7 goes out, which is another reason I'd advocate for using `isProductVersionCompatible`.